### PR TITLE
feat: make ir deployments default

### PR DESCRIPTION
**Pull request by raulb**
## Description of change

Fixes https://github.com/meroxa/product/issues/559

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

**Before**

Regular deployment (pre: intermediate representation)

```
$ meroxa apps deploy .
```

To use a deployment using Intermediate representation:

```
$ meroxa apps deploy . --spec 0.1.1
```

**After**

To use a deployment using Intermediate representation, it'll fetch latest version and it'll be sent to all turbine libraries.

```
$ meroxa apps deploy .
```

## Additional references

This will be a nice to have right after we ship this pull-request to prevent unexpected behavior coming from old libraries: https://github.com/meroxa/product/issues/598.

A changelog will be needed in https://github.com/meroxa/product/issues/593.

### CODE REVIEW

File: cmd/meroxa/turbine/golang/deploy.go [hunk 1]
```diff
@@ -52,13 +50,11 @@ func (t *turbineGoCLI) Deploy(ctx context.Context, imageName, appName, gitSha, s
...
-			err = fmt.Errorf(
-				"unable to receive the deployment spec for the Meroxa Application at %s", t.appPath)
-		}
+	deploymentSpec, err = utils.GetTurbineResponseFromOutput(string(output))
+	if err != nil {
+		err = fmt.Errorf("unable to receive the deployment spec for the Meroxa Application at %s", t.appPath)
 	}
+
 	return deploymentSpec, err
 }
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/golang/deploy.go:59]: I believe you can also remove this branching point:
https://github.com/meroxa/cli/blob/c4e2eb0acff3d9db3dbadbc882c9961c95e3f693/cmd/meroxa/turbine/golang/deploy.go#L119-L123

It was there before to support the old cli

**Review by dianadoherty — APPROVED** [review]
Looks great! Thank you for your work :)

File: cmd/meroxa/turbine/golang/deploy.go (hunk 1 above)
Comment (raulb) [reply cmd/meroxa/turbine/golang/deploy.go:59]: @dianadoherty this is for building which is something unrelated to IR deployments. I think this is still relevant.

Comment (raulb) [reply cmd/meroxa/turbine/golang/deploy.go:62]: Update: this change was technically added unrelated to IR (https://github.com/meroxa/turbine-go/pull/88). We should still support this way of parsing turbine's library output for those running older versions of turbine.

**Pull request merged**